src/test/java/fi/metatavu/soteapi/wnspusher/WnsPusherNotificationControllerTest.java

```diff
@@ -0,0 +1,143 @@
+
+    Supplier<CloseableHttpClient> httpClientSupplier = () -> httpClient;
+    
+    subject = new WnsPusherNotificationController(
+        systemSettingController,
+        logger,
+        httpClientSupplier
+    );
+    
+    Mockito
+      .when(systemSettingController.getSettingValue(WnsPusherNotificationConsts.ENDPOINT_SETTING))
+      .thenReturn("TEST_ENDPOINT");
+
+    Mockito
+      .when(systemSettingController.getSettingValue(WnsPusherNotificationConsts.API_KEY_SETTING))
+      .thenReturn("TEST_API_KEY");
+
+    Mockito
+      .when(systemSettingController.getSettingValue(WnsPusherNotificationConsts.APP_SETTING))
+      .thenReturn("TEST_APP");
+
+    Mockito
+      .when(httpClient.execute(httpClientArguments.capture()))
+      .thenReturn(httpResponse);
+    
+    StatusLine statusLine = Mockito.mock(StatusLine.class);
+    Mockito.when(statusLine.getStatusCode()).thenReturn(200);
+    Mockito
+      .when(httpResponse.getStatusLine())
+      .thenReturn(statusLine);
+  }
+
+  @Test
+  public void testSendNotificationToTopic() {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+  }
+
+  @Test
+  public void testSendNotificationHeader() {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    assertEquals(httpClientArguments
+                  .getValue()
+                  .getLastHeader("Authorization")
+                  .getValue(),
+                 "APIKEY TEST_API_KEY");
+  }
+
+  @Test
+  public void testSendNotificationApp()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    ObjectMapper objectMapper = new ObjectMapper();
+    HttpPost httpPost = httpClientArguments.getValue();
+    JsonNode notification = objectMapper.readTree(EntityUtils.toString(httpPost.getEntity()));
+    assertEquals(notification.get("app").asText(), "TEST_APP");
+  }
+
+  @Test
+  public void testSendNotificationUrl()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    HttpPost httpPost = httpClientArguments.getValue();
+    assertEquals(httpPost.getURI().toString(), "TEST_ENDPOINT");
+  }
+
+  @Test
+  public void testSendNotificationContentType()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
+    subject.sendNotificationToTopic("TEST_TOPIC", "TEST_TITLE", "TEST_MESSAGE");
+    ObjectMapper objectMapper = new ObjectMapper();
+    HttpPost httpPost = httpClientArguments.getValue();
+    JsonNode notification = objectMapper.readTree(EntityUtils.toString(httpPost.getEntity()));
+    assertEquals(notification.get("contentType").asText(), "application/xml");
+  }
+
+  @Test
+  public void testSendNotificationWnsType()
+      throws JsonParseException, JsonMappingException, ParseException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)